docs/source/architecture/global_state.rst

```diff
@@ -0,0 +1,99 @@
+
+The state is split into namespaces which allow flexibility for
+transaction family authors to define, share, and reuse global state data
+between transaction processors.
+
+Radix Merkle Tree Overview
+==========================
+
+Merkle Hashes
+-------------
+
+Sawtooth Lake uses an addressable Radix Merkle tree to store data for
+transaction families. Let's break that down: The tree is a Merkle tree because
+it is a copy-on-write data structure which stores successive node hashes
+from leaf-to-root upon any changes to the tree. For a given set of state
+transitions associated with a block, we can generate a single root hash
+which points to that *version* of the tree. By placing this state root
+hash on the block header, we can gain consensus on the expected version
+of state *in addition to* the consensus on the chain of blocks. If a
+validator's state transitions for a block result in a different hash,
+the block is not considered valid. For more information about general
+concepts, see the Merkle_ page on wikipedia.
+
+.. image:: ../images/state_merkle_hashes.*
+   :width: 80%
+   :align: center
+   :alt: Merkle hash diagram
+
+.. _Merkle: https://en.wikipedia.org/wiki/Merkle_tree
+
+Radix Addresses
+---------------
+
+.. image:: ../images/state_address_format.*
+   :width: 80%
+   :align: center
+   :alt: Tree address format
+
+The tree is an addressable Radix tree because addresses uniquely
+identify the paths to leaf nodes in the tree where information is
+stored. An address is a hex-encoded 70 character string representing
+35 bytes. In the tree implementation, each byte is a Radix path segment which
+identifies the next node in the path to the leaf containing the data
+associated with the address. The address format contains a 3 byte
+(6 hex character) namespace prefix which provides 2\ :sup:`24`
+(16,777,216) possible different namespaces in a given instance of
+Sawtooth Lake. The remaining 32 bytes (64 hex characters) are encoded
+based on the specifications of the designer of the namespace, and may
+include schemes for subdividing further, distinguising object types,
+and mapping domain-specific unique identifiers into portions of the address.
+For more information about general concepts, see the Radix_ page on
+wikipedia.
+
+.. image:: ../images/state_radix.*
+   :width: 80%
+   :align: center
+   :alt: Radix addressing diagram
+
+.. _Radix: https://en.wikipedia.org/wiki/Radix_tree
+
+Serialization Concerns
+======================
+
+In addition to questions regarding the encoding of addresses,
+namespace designers also need to define the mechanism of serialization
+and the rules for serializing/deserializing the data stored at addresses.
+The domain-specific Transaction Processor will be making get(address) and
+set(address, data) calls against a version of state that the validator
+provides. get(address) will return the byte array found at that address
+and set(address, data) will set the byte array stored at that address.
```

Comment:
I think these sentences switch tense which threw me off while reading. Should it be "The domain-specific Transaction Processor makes get(address) and set(address, data) calls against a version of state that the validator provides. get(address) returns the byte array found at that address and set(address, data) sets the byte array stored at that address."?